packages/block-editor/src/store/actions.js

```diff
@@ -122,35 +132,37 @@ export function selectBlock( clientId, initialPosition = null ) {
  * Yields action objects used in signalling that the block preceding the given
  * clientId should be selected.
  *
- * @param {string} clientId Block client ID.
+ * @param {string}                  clientId        Block client ID.
+ * @param {?WPBlockInitialPosition} initialPosition Optional initial position.
  */
-export function* selectPreviousBlock( clientId ) {
+export function* selectPreviousBlock( clientId, initialPosition = -1 ) {
```

Comment:
I'm not super excited to see more use of `initialPosition`... Ideally, we should try to get rid of it and directly set the correct selection either in the selection store or the DOM. `initialPosition` is already reduced to only be needed for block deletion. I was hoping that soon we'd be able to get rid of it entirely.

Reply:
> I'm not super excited to see more use of `initialPosition`... Ideally, we should try to get rid of it and directly set the correct selection either in the selection store or the DOM. `initialPosition` is already reduced to only be needed for block deletion. I was hoping that soon we'd be able to get rid of it entirely.

In this model, how would you foresee doing something like we're doing here (and also doing in WritingFlow) where we want to map a specific `DOMRect` offset to a selection?